Add and initial run of inclusive naming action

petesfrench opened the pull request:
## Done

- Adds inclusive naming github action
- Remove any previously existing non-inclusive language
- **NOTE**: Let me know if you disagree with the changes or you have another suggestion

## QA

- Check github action works
- Check language changes make lexical sense


## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/4451

webteam-app commented:
Demo starting at https://maas-io-613.demos.haus

billwear commented:
I agree that slave is definitely offensive and, IMHO, should be changed.  Native does not bother me, and it feels like its pejorative connotations are more contextual.  Changing "bond" makes no sense to me, if that's what's being recommended.  "Secondary" is highlighted, but I can't tell if it's marked as an offensive term or an acceptable term. I recommend changing master/slave to "controller" and "agent," which is the most descriptive language I've seen (in Jenkins).  Leave "bond" and "native" in this context.

sparkiegeek commented:
API docs should not be edited in this repo, this is not the source

petesfrench commented:
@billwear Hi and thanks for your feedback. Not sure why "bond" was changed, but I have reverted that. I agree "native" should not be included in this filter and have now removed it from the checks. Are you recommending controller/agent as a term to replace all master/slave references? It would be great to get your opinion [here](https://docs.google.com/document/d/1mJUa1VJHOMWa723dmKmNjSKGW-nlBv9xktzGZZwacVo/edit#) where we discuss name changes.

petesfrench commented:
@sparkiegeek Where should these changes be made if not here?

sparkiegeek commented:
See https://github.com/canonical-web-and-design/maas.io/issues/448

Unfortunately i do not believe a GitHub action is appropriate for this repo for the very reason that it contains non-source material

anthonydillon commented:
The GitHub action only checks changes so the API docs will fail when updated but we should be aware of this and working to rectify it going forward. 

The check can be by passed if justified but should be a recurring reminder.

@petesfrench can you revert the API docs changes.

anthonydillon commented:
@sparkiegeek why is this blocked?

petesfrench commented:
@sparkiegeek I made the changes based on our call, let me know if it good to be merged once it has been reviewed.

bethcollins92 reviewed (approved):
woooo nice one!
